lib/handler/file.c

```diff
@@ -303,7 +331,7 @@ static void do_send_file(struct st_h2o_sendfile_generator_t *self, h2o_req_t *re
     /* setup response */
     req->res.status = status;
     req->res.reason = reason;
-    req->res.content_length = self->bytesleft;
+    req->res.content_length = self->gunzip ? SIZE_MAX : self->bytesleft;
```

Comment:
I believe there is an optional attribute in a gzip file that contains the size of the original content, and that the attribute is accessible from libz. It would be beneficial to use the information, since the clients that lack gzip support would likely overlap with those that do not support chunked encoding (i.e. HTTP/1.0).